packages/http-server/test/integration/http-server.integration.ts

```diff
@@ -184,18 +195,51 @@ describe('HttpServer (integration)', () => {
 
   it('converts host from [::] to [::1] in url', async () => {
     // Safari on MacOS does not support http://[::]:3000/
-    server = new HttpServer(dummyRequestHandler, {host: '::'});
+    server = givenHttpServer(dummyRequestHandler, {host: '::'});
     await server.start();
     expect(server.url).to.equal(`http://[::1]:${server.port}`);
   });
 
   it('converts host from 0.0.0.0 to 127.0.0.1 in url', async () => {
     // Windows does not support http://0.0.0.0:3000/
-    server = new HttpServer(dummyRequestHandler, {host: '0.0.0.0'});
+    server = givenHttpServer(dummyRequestHandler, {host: '0.0.0.0'});
     await server.start();
     expect(server.url).to.equal(`http://127.0.0.1:${server.port}`);
   });
 
+  it('supports HTTP/2 protocol with key and certificate files', async () => {
+    // For some reason, spdy fails with Node 11
+    if (+process.versions.node.split('.')[0] > 10) return;
+    const serverOptions: Http2Options = Object.assign(
+      {
+        protocol: 'http2' as 'http2',
+        rejectUnauthorized: false,
+        spdy: {protocols: ['h2' as 'h2']},
+      },
+      givenHttpServerConfig(),
+    );
+    server = givenHttp2Server(serverOptions);
+    await server.start();
+    // http2 does not have its own url scheme
+    expect(server.url).to.match(/^https\:/);
+    const agent = spdy.createAgent({
+      rejectUnauthorized: false,
+      port: server.port,
+      host: server.host,
+      // Optional SPDY options
+      spdy: {
+        plain: false,
+        ssl: true,
+      },
+    }) as spdy.Agent;
+    const response = await httpsGetAsync(server.url, agent);
+    expect(response.statusCode).to.equal(200);
+    // We need to close the agent so that server.close() returns
+    // `@types/spdy@3.x` is not fully compatible with `spdy@4.0.0`
+    // tslint:disable-next-line:no-any
+    (agent as any).close();
```

Comment:
Please close the agent before making assertion.

In your current version, when `expect(response.statusCode)` throw an assertion error, then the agent is never closed.